### ISSUE TITLE
[spike] Define implementation to work towards reducing dataset processing failures in 2019Q3

### ISSUE THREAD

**Issue by morrisonnorman** [HumanCellAtlas/dcp#423]
[Slack channel](https://humancellatlas.slack.com/archives/CM00ETE8L)
[Working document](https://docs.google.com/document/d/1NeYEWkOj25xdV4hM96q-v-0DLOZynmLnWPqlMI3Tjtk)

**Comment by parthshahva** [comment]
@justincc are you leading this spike? Is this going to be involve updating our scale testing strategy and analysis of current state of error reporting? Are we aware of where the gaps are currently?

**Comment by justincc** [comment]
@parthshahva I will lead this spike. I think this will involve updating our scale testing and error reporting if that will fit in this quarter. Either way your help will be crucial. We'll need a rough plan agreed in conjunction with the rest of the involved components (ingest, upload, datastore, analysis) before we can go away and estimate. As discussed elsewhere, it makes sense for us to at least have an initial convo today or tomorrow to get a very draft outline of the basics then start discussions with everyone else. However, if anyone also wants to get in at that super-early stage please say.

**Comment by justincc** [comment]
Note: this issues incorporates both identified per-component reliability work *and* DCP-wide work to improve end-to-end reliability.

**Comment by justincc** [comment]
[Working document](https://docs.google.com/document/d/1NeYEWkOj25xdV4hM96q-v-0DLOZynmLnWPqlMI3Tjtk/edit). Working group to be set up soon to refine.

**Comment by justincc** [comment]
In the 2019-08-01 backlog meeting we discussed that this work will certainly take more than Q3. I agreed to reach consensus in the reliability wg as to what we wanted in scope for Q3, with other work being put into a Q4 epic or beyond.

**Comment by justincc** [comment]
A probable Q3 scope is:

a) Intra-component: Each component identifies per-component work that they believe can be performed in Q3. Ingest has [an epic](https://app.zenhub.com/workspaces/dcp-5ac7bcf9465cb172b77760d9/issues/humancellatlas/ingest-central/433) but this has not been winnowed down into what is achievable. Other components have not identified necessary Q3 reliability work, assuming they think they have any.

b) Inter-component: [We agree, determine, report, measure and track 2 metrics in a crude fashion](https://docs.google.com/document/d/1NeYEWkOj25xdV4hM96q-v-0DLOZynmLnWPqlMI3Tjtk/edit#heading=h.nrnoqqg9xe5o). This work has not yet been sized.

The steps beyond Q3 rely on what actually works in Q3. However, I strongly believe that INSANE reliability work cannot stop. As I see it, either we:
a) Accept that in Q4 we have to work in an agile rather than waterfall fashion on this
b) Come up with guesses that are extremely like to be wrong. Even if we do this we should not spend a lot of time on these guesses since more time spent will not improve their accuracy.

**Comment by justincc** [comment]
No objections have been made to reliability 2019Q3 scope so I am calling this task done.

**Comment by mweiden** [comment]
@justincc Per our AI in the refinement session, could you and @morrisonnorman link the Q4+ work into a future epic so that we have an idea of the scope for what remains in Q3 and what is future work?
